iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java

```diff
@@ -217,15 +215,14 @@ private static void writeExtensions(Network n, NetworkXmlWriterContext context,
 
             Collection<? extends Extension<? extends Identifiable<?>>> extensions = identifiable.getExtensions().stream()
                     .filter(e -> options.withExtension(e.getName()))
+                    .filter(e -> !e.isEmpty())
```

Comment:
I think we really need to cache the extension list for an identifiable, to not iterate several times over the same list.

Reply:
Indeed, it's done twice when writing the xml: first for the namespaces, then for the extension itslef

Reply:
It's now computed only once, at the beginning of the `NetworkXml.write()` static method.